components/cookie-jar-balance.tsx

```diff
@@ -39,7 +44,24 @@ const CookieJarBalance = ({
         </p>
       )}
       <p>Claim period: {periodLength}</p>
-      <p>Owned by: {truncateEthereumAddress(owner)}</p>
+      <div className="flex flex-row gap-x-2">
+        <span>Owned by: {truncateEthereumAddress(owner)} </span>
+        <Button
+          variant="ghost"
+          size="icon"
+          className="h-6 w-6 text-gray-400 hover:text-white"
+          aria-label="Copy address"
+          onClick={() => {
+            navigator.clipboard.writeText(owner);
+            toast({
+              variant: "default",
+              description: `Address copied!`,
+            });
+          }}
+        >
```

Comment:
It is counter-intuitive, but the target of the CookieJar is a Gnosis Safe controlled by the jar. The CookieJar itself is an ERC6551, and the funds live outside the jar.
```suggestion
        <Button
          variant="ghost"
          size="icon"
          className="h-6 w-6 text-gray-400 hover:text-white"
          aria-label="Copy address"
          onClick={() => {
            navigator.clipboard.writeText(jarTargetAddress);
            toast({
              variant: "default",
              description: `Address copied!`,
            });
          }}
        >
```